File: Problems/MinimumCostToConnectSticks.py
import heapq
from typing import List


def connectSticks(sticks: List[int]) -> int:
    if len(sticks) == 1:
        return 0
    heapq.heapify(sticks)
    res = 0
    while len(sticks) > 1:
        first = heapq.heappop(sticks)
        second = heapq.heappop(sticks)
        sum = first + second
        res += sum
        heapq.heappush(sticks, sum)
    return res


print(connectSticks([1, 8, 3, 5]))

File: Problems/MinimumCostToConnectSticks.cpp
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

class Solution
{
public:
    int connectSticks(vector<int> &sticks)
    {
        priority_queue<int, vector<int>, greater<int>> pq(begin(sticks), end(sticks));
        int res = 0;
        while (pq.size() > 1)
        {
            int sum = pq.top();
            pq.pop();
            sum += pq.top();
            pq.pop();
            res += sum;
            pq.push(sum);
        }
        return res;
    }
};